JotunnLib/Entities/CustomRoom.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            Prefab = prefab;
+            Name = prefab.name;
+            if (prefab != null && prefab.TryGetComponent<Room>(out var room))
+            {
+                Room = room;
+            }
+            else
+            {
+                Room = prefab.AddComponent<Room>();
+            }
+            FixReference = fixReference;
+            ApplyConfig(roomConfig);
+        }
+
+        /// <summary>
+        ///     Helper method to determine if a prefab with a given name is a custom room created with Jötunn.
+        /// </summary>
+        /// <param name="prefabName">Name of the prefab to test.</param>
+        /// <returns>true if the prefab is added as a custom item to the <see cref="DungeonManager"/>.</returns>
+        public static bool IsCustomRoom(string prefabName)
+        {
+            return DungeonManager.Instance.Rooms.ContainsKey(prefabName);
+        }
+
+        private void ApplyConfig(RoomConfig roomConfig)
+        {
+            // Ensure Room values are overwritten only when a value's present.
+            if (roomConfig.Enabled != null) Room.m_enabled = roomConfig.Enabled.Value;
+            if (roomConfig.Entrance != null) Room.m_entrance = roomConfig.Entrance.Value;
+            if (roomConfig.Endcap != null) Room.m_endCap = roomConfig.Endcap.Value;
+            if (roomConfig.Divider != null) Room.m_divider = roomConfig.Divider.Value;
+            if (roomConfig.EndcapPrio != null) Room.m_endCapPrio = roomConfig.EndcapPrio.Value;
+            if (roomConfig.MinPlaceOrder != null) Room.m_minPlaceOrder = roomConfig.MinPlaceOrder.Value;
+            if (roomConfig.Weight != null) Room.m_weight = roomConfig.Weight.Value;
+            if (roomConfig.FaceCenter != null) Room.m_faceCenter = roomConfig.FaceCenter.Value;
+            if (roomConfig.Perimeter != null) Room.m_perimeter = roomConfig.Perimeter.Value;
+            if (roomConfig.MusicPrefab != null) Room.m_musicPrefab = roomConfig.MusicPrefab;
+
+            // Room can be matched by either a Room.Theme flag, or a ThemeName string.
+            if (roomConfig.Theme == 0)
+            {
+                ThemeName = roomConfig.ThemeName;
+                Theme = 0;
+            }
+            else
+            {
+                Theme = roomConfig.Theme;
+                ThemeName = string.Empty;
+            }
+
+            // DungeonGenerator.PlaceRoom*() utilize soft references directly, thus registering the assets here.
+            RoomData = new DungeonDB.RoomData()
+            {
+                m_prefab = new SoftReference<GameObject>(AssetManager.Instance.AddAsset(Prefab)),
+                m_loadedRoom = Room,
+                m_prefabData = new RoomPrefabData()
+                {
+                    m_enabled = Room.m_enabled,
+                    m_theme = Theme
+                }
+            };
+        }
```

Comment:
Apply methods are usually located in the config classes, not the manager directly. For consistency this should be moved.

Reply:
Fixed for real in 47472f1df77d941d7f7afbb86f56e21ad6cfc35e